ska_helpers/utils.py

```diff
@@ -375,26 +375,36 @@ def convert_to_int_float_str(val: str) -> int | float | str:
 
 
 class TypedDescriptor:
-    """Class to create a descriptor for an attribute that is cast to a type.
+    """Class to create a descriptor for a dataclass attribute that is cast to a type.
 
     This is a base class for creating a descriptor that can be used to define an
-    attribute on a class that is cast to a specific type.  The type is specified by
+    attribute on a dataclass that is cast to a specific type.  The type is specified by
```

Comment:
@javierggt - you are right this can work in a normal class. The issue there is that it is easy to make a mistake. Your example does not work if you pass in a `time=value` to the class constructor. It should be:
```
In [10]: class MyClass_1:
    ...:     time: CxoTime = CxoTimeDescriptor(default=CxoTime.NOW)
    ...: 
    ...:     def __init__(self, time=None, **kwargs):
    ...:         self.time = time or MyClass_1.time
```
The point of dataclasses is to reduce the boilerplate in the `__init__`. I'm moving to more concrete classes where all the attributes are well-defined from the start, so dataclasses are making a lot of sense.

For the documentation in this case I think less is more. Experts who understand all the machinery don't need the documentation examples, and the rest should only use this in a dataclass as the intended and tested use-case.